batcher_test.go

```diff
@@ -50,105 +51,85 @@ func TestPGXBatcher_Execute_Errors(t *testing.T) {
 			t.Errorf("Expected 1 error, but got %d", len(errs))
 		}
 
-		if errs[0].sql != "INVALID SQL" {
+		/*if errs[0].sql != "INVALID SQL" {
 			t.Errorf("Expected error SQL to be 'INVALID SQL', but got '%s'", errs[0].sql)
-		}
+		}*/
 	} else {
 		t.Errorf("Expected error of type 'BatcherErrors', but got '%T'", err)
 	}
 }
 
 func TestPGXBatcher_Execute_Transactional_Errors(t *testing.T) {
-
-	batcher := New(conn, true)
+	b := New(conn, true)
 
 	// add invalid SQL statement to batch
-	batcher.Queue("INVALID SQL", []interface{}{})
+	b.Queue("INVALID SQL")
 
 	// execute batch
-	err := batcher.Execute(context.TODO())
+	err := b.Execute(context.TODO())
 
 	// assert error type and message
 	if err == nil {
 		t.Error("Expected error, but got nil")
 	}
 
 	if errs, ok := err.(StatementErrors); ok {
-		if len(errs) != 1 {
-			t.Errorf("Expected 1 error, but got %d", len(errs))
+		if len(errs) != 3 {
+			t.Errorf("Expected 3 error, but got %d: \n%v", len(errs), errs.Error())
 		}
 
-		if errs[0].sql != "INVALID SQL" {
+		/*if errs[0].sql != "INVALID SQL" {
 			t.Errorf("Expected error SQL to be 'INVALID SQL', but got '%s'", errs[0].sql)
-		}
+		}*/
 	} else {
 		t.Errorf("Expected error of type 'BatcherErrors', but got '%T'", err)
 	}
 }
 
 func TestPGXBatcher_Reset(t *testing.T) {
-	batcher := New(conn, false)
+	b := New(conn, false)
 
 	// Queue some queries
-	batcher.Queue("INSERT INTO users (name, email) VALUES ($1, $2)", []interface{}{"Alice", "alice@example.com"})
-	batcher.Queue("INSERT INTO users (name, email) VALUES ($1, $2)", []interface{}{"Bob", "bob@example.com"})
+	b.Queue("INSERT INTO users (name, email) VALUES ($1, $2)", []interface{}{"Alice", "alice@example.com"})
+	b.Queue("INSERT INTO users (name, email) VALUES ($1, $2)", []interface{}{"Bob", "bob@example.com"})
 
 	// Reset the batcher
-	batcher.Reset()
+	b.Reset()
 
 	// Ensure that the batch is empty after reset
-	if len(batcher.queries) != 0 {
-		t.Errorf("Expected empty batch after Reset(), got %+v", batcher.queries)
+	if b.batch.Len() != 0 {
+		t.Errorf("Expected empty batch after Reset(), got %+v", b.batch.Len())
 		t.Fail()
 	}
 
 	// execute batch
-	err := batcher.Execute(context.TODO())
+	err := b.Execute(context.TODO())
 	if err == nil {
 		t.Error("expected batcher will fail with no queries")
 		t.Fail()
 	}
-	if err.Error() != "no queries to execute" {
-		t.Errorf("unexpected error %s", err)
-	}
-}
 
-func TestStatementErrors_Error(t *testing.T) {
```

Comment:
Happy to remove this due to the changes in errors, but it would be nice to test the error changes. It was important to provide the sql to help in diagnosing issues in a batch. (It would have been better I originally used insert sql statements as the example)

But If I had 10 or 50 queries in a batch and I have one issue in a statement, it would be easier to diagnose with a specific error